test/easyconfigs/easyconfigs.py

```diff
@@ -738,8 +758,8 @@ def check_https(self, changed_ecs):
 
                 if https_url_works:
                     failing_checks.append("Found http:// URL in %s, should be https:// : %s" % (ec_fn, http_url))
-
-        self.assertFalse(failing_checks, '\n'.join(failing_checks))
+        if failing_checks:
+            self.fail('\n'.join(failing_checks))
```

Comment:
@Flamefire what was wrong with `self.assertFalse` here?

Reply:
This has always bugged me when having to read a failing CI log.
In short: The `failing_checks` contains error strings like `X should by Y`, if there are any errors, the test should fail and the error messages should be shown. This is what `fail` does. 
While `assertFalse` kinda works too it is simply not the right tool here. The error message would be something like `Test xxx failed: <many error messages> is not False: <many error messages>`. With `fail` it is `Test xxx failed: <many error messages>` I hope it is apparent why the latter is better and less confusing

Reply:
:+1:

Reply:
:+1:  there are a lot of other places where we could make this change